app/pages/4_Clean_Tables.py

```diff
@@ -109,30 +122,41 @@ def style_symbol(v, props=""):
         symbol = st.checkbox(
             "Show the cells that contain a special symbol : " + special_characters
         )
+        remove_symbols = st.checkbox("Remove the special symbols")
+
+    dataframe = edited_df
 
-    if total or negativ or symbol:
-        dataframe = edited_df
-        new_row = edited_df.apply(column_sum, axis=0)
+    if remove_symbols:
+        pattern = "[" + re.escape(special_characters) + "]|\(.*?\)"
```

Comment:
Since the parentheses are also in the `special_characters` and the conditions are processed from left to right, the regex will not consider the second condition for the parentheses that have already been matched by the first condition.
A solution that seems to work is to change the conditions positions: --> `pattern = "\(.*?\)|" + "[" + re.escape(special_characters) + "]"`
Examples:
```
re.findall("\(.*?\)|" + "[" + re.escape(special_characters) + "]", "42 (123)")
>>> ['(123)']
re.findall("[" + re.escape(special_characters) + "]" + "|\(.*?\)", "42 (123)")
['(', ')']
```